{"name": "登录成功【期望结果故意写错】", "status": "failed", "statusDetails": {"message": "Failed: 断言失败: 预期[登录成功], 实际: [SUCCESS]\nassert 'SUCCESS' == '登录成功'\n  \n  - 登录成功\n  + SUCCESS", "trace": "self = <test_case.test_case_002.TestCase object at 0x000001A92A59BAD0>\ncase_data = {'id': 17, 'url': 'http://novel.hctestedu.com', 'path': '/user/login', 'method': 'post', ...}\n\n    @pytest.mark.parametrize(\"case_data\", CaseData)\n    def testcase(self, case_data: Dict[str, Any]):\n        self.__dynamic_title(case_data)\n        case_data = eval(Template(str(case_data)).render(self.all_dict))\n    \n        test_result = \"Passed\"  # 默认状态\n        error_msg = None\n    \n        try:\n    \n            # 1. 准备请求数据\n            dict_data = self.__prepare_request_data(case_data)\n            print(dict_data)\n            # 2. 发送请求\n            res = getattr(self.ak, case_data[\"method\"])(**dict_data)\n            res_json = res.json()\n            print(res_json)\n    \n            # 3. 获取实际结果并断言\n            actual_result = self.ak.get_value(res_json, case_data[\"actualResult\"])\n            print(f\"actual_result: {actual_result}\")\n>           assert actual_result == case_data[\"expectResult\"], \\\n                f\"断言失败: 预期[{case_data['expectResult']}], 实际: [{actual_result}]\"\nE               AssertionError: 断言失败: 预期[登录成功], 实际: [SUCCESS]\nE               assert 'SUCCESS' == '登录成功'\nE                 \nE                 - 登录成功\nE                 + SUCCESS\n\ntest_case\\test_case_002.py:141: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_case.test_case_002.TestCase object at 0x000001A92A59BAD0>\ncase_data = {'id': 17, 'url': 'http://novel.hctestedu.com', 'path': '/user/login', 'method': 'post', ...}\n\n    @pytest.mark.parametrize(\"case_data\", CaseData)\n    def testcase(self, case_data: Dict[str, Any]):\n        self.__dynamic_title(case_data)\n        case_data = eval(Template(str(case_data)).render(self.all_dict))\n    \n        test_result = \"Passed\"  # 默认状态\n        error_msg = None\n    \n        try:\n    \n            # 1. 准备请求数据\n            dict_data = self.__prepare_request_data(case_data)\n            print(dict_data)\n            # 2. 发送请求\n            res = getattr(self.ak, case_data[\"method\"])(**dict_data)\n            res_json = res.json()\n            print(res_json)\n    \n            # 3. 获取实际结果并断言\n            actual_result = self.ak.get_value(res_json, case_data[\"actualResult\"])\n            print(f\"actual_result: {actual_result}\")\n            assert actual_result == case_data[\"expectResult\"], \\\n                f\"断言失败: 预期[{case_data['expectResult']}], 实际: [{actual_result}]\"\n            self.__json_extraction(case_data, res_json)\n            self.__sql_assert(case_data)\n    \n            if case_data[\"responseExpect\"]:\n                deepdiff_expect_data = self.__safe_json_parse(case_data.get(\"responseExpect\"))\n                other = eval(case_data[\"responseExclude\"])\n                print(f\"得到的responseExclude: {other}\")\n                jsonMaxDataRes = self.ak.data_deepdiff(deepdiff_expect_data, res_json, **other)\n                assert  jsonMaxDataRes, \"大量数据响应断言失败，两者数据不一致\"\n    \n        except AssertionError as ae:\n            # 断言失败的专属处理\n            test_result = \"Failed\"\n            error_msg = str(ae)\n>           pytest.fail(error_msg)  # 标记测试失败\n            ^^^^^^^^^^^^^^^^^^^^^^\nE           Failed: 断言失败: 预期[登录成功], 实际: [SUCCESS]\nE           assert 'SUCCESS' == '登录成功'\nE             \nE             - 登录成功\nE             + SUCCESS\n\ntest_case\\test_case_002.py:157: Failed"}, "description": "用例登陆", "steps": [{"name": "发送post请求", "status": "passed", "parameters": [{"name": "url", "value": "'http://novel.hctestedu.com/user/login'"}, {"name": "data", "value": "{'username': '15096261110', 'password': 'asd123...'}"}, {"name": "json", "value": "None"}, {"name": "params", "value": "{}"}, {"name": "headers", "value": "{}"}], "start": 1756101450308, "stop": 1756101450365}, {"name": "jsonpath提取数据", "status": "passed", "parameters": [{"name": "data", "value": "{'code': '200', 'msg': 'SUCCESS', 'data': {'token': 'eyJhbGciOiJIUzUxMiJ9.eyJleHAiOjE3NTY3MDYyNTAsInN1YiI6IntcImlkXCI6MTYyODc0MDY3ODY3MDE5NjczNixcInVzZXJuYW1lXCI6XCIxNTA5NjI2MTExMFwiLFwibmlja05hbWVcIjpcIlMxNTA5NjI2MTExMFwifSIsImNyZWF0ZWQiOjE3NTYxMDE0NTAzODh9.FrrFn3180y3lpnJ-Yz0tRJvmsmnTyLZgowKopdb7QwVyXMVZme2epLDxKHvcdGl08rQJOCJp3ghLUCFP-_RUcA'}}"}, {"name": "key", "value": "'$.msg'"}], "start": 1756101450366, "stop": 1756101450366}], "attachments": [{"name": "log", "source": "b7a41158-a787-430a-b988-c57e550bbe1e-attachment.txt", "type": "text/plain"}, {"name": "stdout", "source": "1d99da32-8885-4777-9128-4488a334cf8f-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "case_data", "value": "{'id': 17, 'url': 'http://novel.hctestedu.com', 'path': '/user/login', 'method': 'post', 'params': None, 'headers': None, 'data': '{\"username\":\"15096261110\",\"password\":\"asd123...\"}', 'type': 'data', 'actualResult': '$.msg', 'expectResult': '登录成功', 'result': \"Failed: 断言失败: 预期[登录成功], 实际: [SUCCESS]\\nassert 'SUCCESS' == '登录成功'\\n  \\n  - 登录成功\\n  + SUCCESS\", 'caseName': '登录成功【期望结果故意写错】', 'jsonExData': None, 'sqlExData': None, 'regularExData': None, 'storyName': 'F01S01_登陆', 'featureName': 'F05_读书屋项目', 'remark': '用例登陆', 'rank': 'blocker', 'sqlAssertData': None, 'sqlExpectResult': None, 'responseExpect': None, 'responseExclude': None, 'is_true': True, None: None}"}], "start": 1756101450307, "stop": 1756101450394, "uuid": "3b5ed43d-5641-4c6e-8a12-5bfb7ab46102", "historyId": "261ef121f71ac105cd9fac1ff1fdcb7b", "testCaseId": "418d268ce69e160b0573de682d1350f4", "fullName": "test_case.test_case_002.TestCase#testcase", "labels": [{"name": "story", "value": "F01S01_登陆"}, {"name": "feature", "value": "F05_读书屋项目"}, {"name": "severity", "value": "blocker"}, {"name": "parentSuite", "value": "test_case"}, {"name": "suite", "value": "test_case_002"}, {"name": "subSuite", "value": "TestCase"}, {"name": "host", "value": "DESKTOP-CQ0OVRO"}, {"name": "thread", "value": "2792-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_case.test_case_002"}], "titlePath": ["test_case", "test_case_002.py", "TestCase"]}